module/surveySubmissions/helper.js

```diff
@@ -0,0 +1,421 @@
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                })
+            }
+        })
+    }
+
+
+     /**
+    * List created and submitted surveys.
+    * @method
+    * @name list
+    * @param {String} userId - logged in userId
+    * @returns {Json} - survey list.
+    */
+
+    static list(userId = "") {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                if (userId == "") {
+                    throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK)
+                }
+
+                let getSurveyList = [
+                    solutionsHelper.solutionDocumentsByAggregateQuery
+                    (
+                            [
+                                {
+                                    "$match": {
+                                        "author": userId,
+                                        "type": messageConstants.common.SURVEY,
+                                        "isDeleted": false,
+                                    }
+                                },
+                                {
+                                    "$project": {
+                                        "solutionId": "$_id",
+                                        "name": 1,
+                                        "status": 1
+
+                                    }
+                                },
+                                { "$sort": { createdAt: 1 } }
+                            ]
+                        ),
+                    database.models.surveySubmissions.aggregate
+                        (
+                            [
+                                { "$match": { "createdBy": userId } },
+                                {
+                                    "$project": {
+                                        'submissionId': "$_id",
+                                        'name': 1,
+                                        "surveyId": 1,
+                                        "solutionId": 1,
+                                        "endDate": 1,
+                                        "status": 1
+
+                                    }
+                                },
+                                { "$sort": { createdAt: 1 } }
+                            ]
+                        )
+                ]
+
+                let result = [];
+
+                await Promise.all(getSurveyList)
+                    .then(function (response) {
+                        
+                        if (response[0].length > 0) {
+                            result = result.concat(response[0]);
+                        }
+
+                        if (response[1].length > 0) {
+                            response[1].forEach( async surveySubmission => {
+                                if (new Date > new Date(surveySubmission.endDate) && 
```

Comment:
@deepakr123 - Start date and end date to be taken from surveyInformation. Also remove name key from surveySubmission

Reply:
resolved